gcloud/_helpers.py

```diff
@@ -159,6 +160,23 @@ def _app_engine_id():
     return app_identity.get_application_id()
 
 
+def _file_project_id():
+    """Gets the project id from the credentials file if one is available.
+
+    :rtype: string or ``NoneType``
+    :returns: Project-ID from JSON credentials file if value exists,
+              else ``None``.
+    """
+    credentials_file_path = os.getenv('GOOGLE_APPLICATION_CREDENTIALS')
+    if credentials_file_path:
+        credentials_file = open(credentials_file_path, 'r')
+        credentials_data = credentials_file.read()
+        credentials = json.loads(credentials_data)
```

Comment:
Please use `with open(credentials_file_path, 'rb') as credentials_file`. You seem to be allergic to context managers?